src/btf.cpp

```diff
@@ -737,51 +720,48 @@ FuncParamLists BTF::get_params_from_btf(
   if (id != (max + 1))
     LOG(ERROR) << "BTF data inconsistency " << id << "," << max;
 
-  btf_dump__free(dump);
-
   return params;
 }
 
 FuncParamLists BTF::get_raw_tracepoints_params_from_btf(
     const BTFObj &btf_obj,
     const std::set<std::string> &rawtracepoints) const
 {
-  __s32 id, max = static_cast<__s32>(type_cnt(btf_obj.btf));
-  std::string type = std::string("");
-  struct btf_dump *dump;
-  char err_buf[256];
-  int err;
-
-  dump = dump_new(btf_obj.btf, dump_printf, &type);
-  err = libbpf_get_error(dump);
-  if (err) {
+  std::string type;
+  auto *dump = dump_new(btf_obj.btf, dump_printf, &type);
+  if (auto err = libbpf_get_error(dump)) {
+    char err_buf[256] = { 0 };
     libbpf_strerror(err, err_buf, sizeof(err_buf));
     LOG(ERROR) << "BTF: failed to initialize dump (" << err_buf << ")";
     return {};
   }
+  SCOPE_EXIT
+  {
+    btf_dump__free(dump);
+  };
 
   FuncParamLists params;
-  for (id = start_id(btf_obj.btf); id <= max; id++) {
+  auto id = start_id(btf_obj.btf), max = type_cnt(btf_obj.btf);
+  for (; id <= max; id++) {
     const struct btf_type *t = btf__type_by_id(btf_obj.btf, id);
-
     if (!t)
       continue;
 
     if (!btf_is_func(t))
       continue;
 
-    const char *str = btf__name_by_offset(btf_obj.btf, t->name_off);
-    std::string tp_name = str;
-
+    std::string_view tp_name = btf__name_by_offset(btf_obj.btf, t->name_off);
     for (const auto &prefix : RT_BTF_PREFIXES) {
-      if (tp_name.starts_with(prefix)) {
-        tp_name.erase(0, prefix.length());
-      }
+      if (tp_name.starts_with(prefix))
+        tp_name.remove_prefix(prefix.length());
     }
-    tp_name = btf_obj.name + ":" + tp_name;
 
     // Checking multiple prefixes so make sure we don't add duplicates
-    if (!rawtracepoints.contains(tp_name) || params.contains(tp_name))
+    auto mod_tp_name = btf_obj.name + ":" + std::string(tp_name); // TODO:
```

Comment:
Looks like this todo comment got messed up here. It's referring to the tp_name right?